snappy/hwaccess.go

```diff
@@ -53,6 +53,18 @@ func validDevice(device string) bool {
 	return false
 }
 
+func validDeviceForUdev(device string) bool {
+	inValidPrefixes := []string{"/sys/devices", "/sys/class/gpio"}
```

Comment:
@clobrano Thanks a lot for your branch and sorry for the slow reply. I like your suggestions for the new names. While the code will change with capabilities my gut-feeling is that its is useful to fix this bug (I assume we will use a similar low-level approach as we do for hw-assign). So +1 for this branch and for applying the cleanups. If you need any help or are busy just let me know and I can do the small renames too :)